src/config/compiler.rs

```diff
@@ -1,7 +1,7 @@
 use super::{builder::ConfigBuilder, validation, Config, TransformOuter};
 use indexmap::IndexMap;
 
-pub fn compile(raw: ConfigBuilder) -> Result<Config, Vec<String>> {
+pub fn compile(raw: ConfigBuilder, warnings: &mut Vec<String>) -> Result<Config, Vec<String>> {
```

Comment:
It was a deliberate choice to go with out parameter to simplify passing of warnings and to avoid adding complexity to the return type. It seamed as less intrusive choice as warning passing is really just for validate command. 

But I'm felling like I'm bike shedding. So yea, we can return it, aliasing could help. 

Regarding the boolean flag, it's a good idea, with that we could avoid returning the warnings. It also needs to be passed to `load_builder_from_paths`. It can also be later extended to denying warnings in components. Let's go with this.